actors/evm/tests/precompile.rs

```diff
@@ -0,0 +1,81 @@
+mod asm;
+
+use evm::interpreter::U256;
+use fil_actor_evm as evm;
+use fil_actors_runtime::test_utils::*;
+use fvm_ipld_encoding::RawBytes;
+
+#[allow(dead_code)]
+pub fn magic_precompile_contract() -> Vec<u8> {
+    let init = r#"
+"#;
+
+    let body = r#"
+push16 0x666F6F206261722062617A20626F7879 # foo bar baz boxy
+push2 0x0100 # offset of input data
+mstore # store value at offset
+
+%push(sha256_hash)
+jump # call hash, output written to 0x0200
+
+sha256_hash:
+jumpdest
+push1 0x20   # out size (32 bytes)
+push2 0x0200 # out offset 
+push1 0x10   # in size (16 bytes)
+push2 0x0110 # in offset
```

Comment:
Wasnt that stored in 0x100?

Reply:
ah the top 16 bytes are 0s, ok. Maybe add a comment here, it made me scratch my head.